rust/src/domains/polars/datetime/mod.rs

```diff
@@ -0,0 +1,18 @@
+use polars::prelude::TimeUnit;
+
+use crate::{core::Domain, error::Fallible};
+
+/// A domain that represents a datetime.
+#[derive(Debug, Clone, PartialEq)]
+pub struct DatetimeDomain {
+    pub time_unit: TimeUnit,
+    pub time_zone: Option<String>,
```

Comment:
This seems odd, but I'm not sure I understand the goal. Expected a composition of Date and Time. Are we expecting to get anything out of the timezone string? Should there be hours ahead or behind UTC?

Reply:
* Time (i64): number of nanoseconds since midnight
* Datetime (i64): number of milli/nano/micro seconds since Unix epoch. Time zone and time unit are encoded as part of the type/shared amongst all rows in a column
* Date (i32): number of days since Unix epoch.

Time and Date don't carry any extra type information/domain descriptors, so they can be represented via AtomDomain.
Datetime is separate as it isn't a composition of date and time, it is its own data format. It is also different in that it has type information (zone and unit) that needs to be known to interpret the meaning of the data. For the purposes of OpenDP, I consider the set of time points spanned by the domain to be distinct for each time zone. Updated the documentation.